src/topology/instances/ennreal.lean

```diff
@@ -663,6 +663,67 @@ begin
   exact ennreal.tendsto_of_real h,
 end
 
+lemma tsum_coe_ne_top_iff_summable_coe {f : α → ℝ≥0} :
+  ∑' a, (f a : ℝ≥0∞) ≠ ∞ ↔ summable (λ a, (f a : ℝ)) :=
+begin
+  rw nnreal.summable_coe,
+  exact tsum_coe_ne_top_iff_summable,
+end
+
+lemma tsum_coe_eq_top_iff_not_summable_coe {f : α → ℝ≥0} :
+  ∑' a, (f a : ℝ≥0∞) = ∞ ↔ ¬ summable (λ a, (f a : ℝ)) :=
+begin
+  rw [← @not_not (∑' a, ↑(f a) = ⊤)],
+  exact not_congr tsum_coe_ne_top_iff_summable_coe
+end
+
+lemma tsum_to_real_eq_of_summable {f : α → ℝ≥0∞} (hf : ∀ a, f a ≠ ∞)
+  (h : summable (ennreal.to_real ∘ f)) :
+  (∑' a, f a).to_real = ∑' a, (f a).to_real :=
+begin
+  obtain ⟨r, hr⟩ := h,
+  rw has_sum.tsum_eq hr,
+  have hr' : (ennreal.of_real r).to_real = r,
+  { rw ennreal.to_real_of_real,
+    rw ← has_sum.tsum_eq hr,
+    apply tsum_nonneg,
+    intro x,
+    exact to_real_nonneg },
+  suffices : has_sum f (ennreal.of_real r),
+  { rw has_sum.tsum_eq this,
+    exact hr' },
+  rw has_sum at *,
+  rw ← tendsto_to_real_iff,
+  { convert hr, ext s,
+    { rw to_real_sum,
+      intros a _,
+      exact lt_top_iff_ne_top.2 (hf a) } },
+  { intros s hs,
+    rw sum_eq_top_iff at hs,
+    obtain ⟨x, _, hx⟩ := hs,
+    exact (hf x) hx },
+  { exact ennreal.of_real_ne_top }
+end
+
+lemma tsum_to_real_eq_of_not_summable {f : ℕ → ℝ≥0∞} (hf : ∀ a, f a ≠ ∞)
+  (h : ¬ summable (ennreal.to_real ∘ f)) :
+  (∑' a, f a).to_real = ∑' a, (f a).to_real :=
+begin
+  lift f to ℕ → ℝ≥0 using hf,
+  change ¬ summable (λ i, (f i : ℝ≥0∞).to_real) at h,
+  simp_rw [coe_to_real] at h ⊢,
+  rw [tsum_eq_zero_of_not_summable h,
+      tsum_coe_eq_top_iff_not_summable_coe.2 h, top_to_real],
+end
+
+lemma tsum_to_real_eq {f : ℕ → ℝ≥0∞} (hf : ∀ a, f a ≠ ∞) :
```

Comment:
Since this lemma strictly supersedes the previous two lemmas, you should make the previous ones private.

Reply:
Done!